doc/maintaining-V8.md

```diff
@@ -0,0 +1,275 @@
+
+The backlog of issues with such is regularly reviewed by the node-team at Google to shepherd through the backport process. External contributors are welcome to collaborate on the backport process as well. Note that some of the bugs may be security issues and will not be visible to external collaborators.
+
+# Updating V8
+
+Node keeps a vendored copy of V8 inside of deps/ directory. In addition Node may need to float patches that do not exist upstream. This means that some care may need to be taken to update the vendored copy of V8.
+
+## Minor updates (patch level)
+
+Because there may floating patches on the version of V8 in Node.js, it is safest to apply the patch level updates as a patch. For example, imagine that upstream V8 is at 5.0.71.47 and Node.js is at 5.0.71.32. It would be best to compute the diff between these tags on the V8 repository, and then apply that patch on the copy of V8 in Node.js. This should preserve the patches/backports that Node.js may be floating (or else cause a merge conflict).
+
+The rough outline of the process is:
+
+```shell
+# Assuming your fork of Node.js is checked out in $NODE_DIR
+# and you want to update the Node.js master branch.
+# Find the current (OLD) version in
+# $NODE_DIR/deps/v8/include/v8-version.h
+cd $NODE_DIR
+git checkout master
+git merge --ff-only upstream/master
+git checkout -b V8_NEW_VERSION
+curl -L https://github.com/v8/v8/compare/${V8_OLD_VERSION}...${V8_NEW_VERSION}.patch > /tmp/patch-1
+cd deps/v8
+patch -p1 < /tmp/patch-1
+cd ../..
+git commit -am 'deps: update V8 from V8_OLD_VERSION to V8_NEW_VERSION'
+```
+
+V8 also keeps tags of the form *5.4-lkgr* which point to the *Last Known Good Release* from the 5.4 branch that can be useful in the update process above.
+
+
+## Major Updates
+
+We upgrade the version of V8 in Node.js master whenever a V8 release goes stable upstream, that is, whenever a new release of Chrome comes out.
+
+Upgrading major versions would be much harder to do with the patch mechanism above. A better strategy is to
+
+1. Audit the current master branch and look at the patches that have been floated since the last major V8 update.
+1. Replace the copy of V8 in Node.js with a fresh checkout of the latest stable V8 branch. Special care must be taken to recursively update the DEPS that V8 has a compile time dependency on (at the moment of this writing, that is only trace_event)
+1. Refloat (cherry-pick) all the patches from list computed in 1) as necessary. Some of the patches may no longer be necessary.
+
+To audit for floating patches:
+
+```shell
+git log --oneline deps/v8
+```
+
+To replace the copy of V8 in Node, use the '[update-v8](https://gist.github.com/targos/8da405e96e98fdff01a395bed365b816)' script<sup>2</sup>. For example, if you want to replace the copy of V8 in Node.js with the branch-head for V8 5.1 branch:
```

Comment:
Yep. @targos is also working on a port with more features. At some point it would be good to make this tooling directly available.